addons/website_sale/views/templates.xml

```diff
@@ -1765,6 +1765,25 @@
                                             <label class="col-form-label" for="phone">Phone</label>
                                             <input type="tel" name="phone" t-attf-class="form-control #{error.get('phone') and 'is-invalid' or ''}" t-att-value="'phone' in checkout and checkout['phone']" />
                                         </div>
+                                        <t t-if="is_view_active('website_sale.address_b2b') or res_company._force_b2b_fields()">
```

Comment:
@nboulif Could you move the method on the website itself and check the is_view_active in there? 
```suggestion
                                        <t t-if="_force_b2b_fields()">
```

Reply:
Also `_force..` might not be appropriate in this case, something more along the lines of `_display_partner_b2b_fields` maybe?